core/memory_tracker/README.md

```diff
@@ -0,0 +1,64 @@
+# Track Coherent Memory
+
+## Problem and Solution
+Coherent memories are implicitly synchronized which means there is no *flush*
+or *invalid* command to explicitly *write* or *read* to/from those memories.
+Henceforth we cannot rely on intercepting the *flush* and *invalid* commands to
+track the modifications for coherent memory.
+
+The solution is to utilize segmentation fault signal handler to tell to which
+memory page has been written and then read the data of that page to track the
+data in coherent memory.
+
+## Goal
+The goal is to implement a memory tracker that:
+* Tracks the memory pages that have been written to.
+* Transparent to the application under debugging.
+ * Assumes zero modification on the application under debugging
+ * For real segmentation faults, the system handler or, if any, the handler
+ registered by the application should be used.
+ * Supports multi-threaded context.
+
+## Current Support
+The current support status of the memory tracker based on operating systems and
+graphics APIs of the tracing target.
+
+|OS        |OpenGL              |Vulkan              |
+|----------|--------------------|--------------------|
+|Windows   |:white_medium_square:|:white_medium_square:|
+|Linux     |:white_medium_square:|:white_check_mark:  |
+|Android   |:white_medium_square:|:white_check_mark:  |
+
+
+## Overall Workflow
+* Start Tracing.
+* Register the memory tracker's segfault handler.
```

Comment:
We only register the segfault handler the first time someone calls "MapMemory" on a coherent buffer